crates/ruma-state-res/src/event_auth.rs

```
@@ -531,12 +541,22 @@ fn valid_membership_change(
             let no_more_prev_events = prev_events.next().is_none();
 
             if prev_event_is_create_event && no_more_prev_events {
-                let create_content =
-                    from_json_str::<RoomCreateEventContent>(create_room.content().get())?;
+                let is_creator = if room_version.use_room_create_sender {
+                    let creator = create_room.sender();
+
+                    creator == sender && creator == target_user
+                } else {
+                    #[allow(deprecated)]
+                    let creator = from_json_str::<RoomCreateEventContent>(
+                        create_room.content().get(),
+                    )?
+                    .creator
+                    .expect("The m.room.create event has a creator in room version 1 through 10");
```

Is it really appropriate to panic here, instead of returning an error?

---

Maybe not, you're right. I forgot that even if the `m.room.create` event was deemed wrong before, we could still be asked to auth an event that depends on it.

---

Returns a serde error now